fhd_core/gridding/gaussian_grid.pro

```diff
@@ -0,0 +1,97 @@
+FUNCTION gaussian_grid,vis_arr_use,obs,psf,params,xcen=xcen,ycen=ycen,vis_inds_use=vis_inds_use,xmin=xmin,$
+  model_use=model_use,model_return=model_return,weights=weights,variance=variance,polarization=polarization,$
+  fi_use=fi_use,n_vis=n_vis,vis_weights=vis_weights,_Extra=extra
+
+  heap_gc
+  i = Complex(0,1)
+
+  ;extract information from the structures
+  p=*psf.beam_gaussian_params[polarization]
+  var = reform(p,5,N_elements(p)/5)
+  dimension=Long(obs.dimension)
+  elements=Long(obs.elements)
+
+  frequency_array=((*obs.baseline_info).freq)[fi_use]
+  n_f_use=N_Elements(fi_use)
+
+  kx = ((FINDGEN(dimension)-dimension/2.)#(FLTARR(elements)+1.))
+  kx_vec = reform(temporary(kx),dimension*elements)
+  ky = ((FLTARR(dimension)+1.)#(FINDGEN(elements)-elements/2.))
+  ky_vec = reform(temporary(ky),dimension*elements)
+
+  weights_vec=complex(FLTARR(dimension*elements))
+  variance_vec=complex(FLTARR(dimension*elements))
+  model_vec=complex(FLTARR(dimension*elements))
+  data_vec=complex(FLTARR(dimension*elements))
+
+  ;Define frequently used constants outside loops for speed
+  normi = 2.*!dpi/(dimension*elements)
+  amp_norm = float(reform(var[0,*]*2.*!dpi/(dimension*elements)*var[2,*]*var[4,*]))
```

Comment:
Minor detail, but you might want to use `normi` here instead of 2.*!dpi/(dimension*elements) again